lib/Route.js

```diff
@@ -181,7 +195,14 @@ class Route {
     }
     try {
       request.routeParams = this._parsePath(request.path);
-      return Q(this.handler(options));
+      if (this.errorHandler) {
+        return Q(Q(this.handler(options)).catch((error) => {
+          return this.errorHandler({ request, response, error });
+        }));
+      }
+      else {
+        return Q(this.handler(options));
```

Comment:
This whole block (lines 198-205) could be cleaned up into a single promise chain.
```
let handlerPromise = Q(this.handler(options));
if (this.errorHandler) {
  handlerPromise = handlerPromise.catch((error) => {
    return this.errorHandler({ request, response, error });
  });
}
return handlerPromise;
```